nix/docker.nix

```diff
@@ -0,0 +1,79 @@
+############################################################################
+# Docker image builder
+#
+# To test it out, use:
+#
+#   docker load -i $(nix-build -A dockerImage --no-out-link)
+#   docker run cardano-node
+#
+############################################################################
+
+{ system ? builtins.currentSystem
+, crossSystem ? null
+, config ? {}
+, profiling ? false
+
+, runtimeShell
+, writeScriptBin
+, dockerTools
+
+, commonLib ? import ../lib.nix { inherit system crossSystem config profiling; }
+
+# The main contents of the image.
+, cardano-node
+
+# Get the current commit
+, gitrev ? commonLib.iohkNix.commitIdFromGitRepoOrZero ../.git
+
+# Other things to include in the image.
+, glibcLocales
+, iana-etc
+, cacert
+, bashInteractive
+, coreutils
+, utillinux
+, iproute
+, iputils
+, curl
+, socat
+
+# Used to generate the docker image names
+, repoName ? "inputoutput/cardano-node"
+}:
+
+let
+
+  # Layer of tools which aren't going to change much between versions.
+  baseImage = dockerTools.buildImage {
+    name = "${repoName}-env";
+    contents = [
+      glibcLocales iana-etc cacert
+      bashInteractive coreutils utillinux iproute iputils curl socat
+    ];
+    # set up /tmp (override with TMPDIR variable)
+    extraCommands = "mkdir -m 0777 tmp";
+  };
+
+in
+  dockerTools.buildImage {
+    name = repoName;
+    ### tag = "${cardano-node.version}";
```

Comment:
Remove this line. The base will always have the git commit as the tag

Reply:
Resolved in commit 5e4d3c8